packages/x-charts/src/internals/defaultValueFormatters.ts

```diff
@@ -10,6 +10,13 @@ export function createScalarFormatter(tickNumber: number, zoomScale: D3Continuou
     context: AxisValueFormatterContext<S>,
   ): string {
     if (context.location === 'tick') {
+      const domain = context.scale.domain();
+      const zeroSizeDomain = domain[0] === domain[1];
+
+      if (zeroSizeDomain) {
+        return `${value}`;
```

Comment:
What about this?

```suggestion
        return context.scale.tickFormat(1)(value);
```

The issue with `${value}` are the dates, of long number formats